// Licensed to the Apache Software Foundation (ASF) under one
// or more contributor license agreements.  See the NOTICE file
// distributed with this work for additional information
// regarding copyright ownership.  The ASF licenses this file
// to you under the Apache License, Version 2.0 (the
// "License"); you may not use this file except in compliance
// with the License.  You may obtain a copy of the License at
//
//   http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing,
// software distributed under the License is distributed on an
// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
// KIND, either express or implied.  See the License for the
// specific language governing permissions and limitations
// under the License.

syntax = "proto2";

package quickstep.serialization;

import "types/Type.proto";

enum LIPFilterType {
  BLOOM_FILTER = 1;
  EXACT_FILTER = 2;
  SINGLE_IDENTITY_HASH_FILTER = 3;
}

message LIPFilter {
  required LIPFilterType lip_filter_type = 1;

  extensions 16 to max;
}

message SingleIdentityHashFilter {
  extend LIPFilter {
    // All required
    optional uint64 num_bits = 16;
  }
}

enum LIPFilterActionType {
  BUILD = 1;
  PROBE = 2;
}

message LIPFilterDeploymentInfo {
  required LIPFilterActionType action_type = 1;
  required uint32 lip_filter_id = 2;
  required int32 attribute_id = 3;
  required Type attribute_type = 4;
}
